p2p/kademlia.py

```diff
@@ -426,7 +426,11 @@ def recv_ping(self, remote: Node, hash_: bytes) -> None:
         reply with a pong, whereas in the latter we'll also fire a callback from ping_callbacks.
         """
         self.logger.debug('<<< ping from %s', remote)
-        self.update_routing_table(remote)
+        if remote == self.this_node:
+            self.logger.debug('Received ping from this_node: %s', remote)
```

Comment:
I wonder how this could happen given that `ping()` refuses to send a ping msg to ourselves. Maybe leave a comment here saying that and that we should investigate this further?